crates/aptos/src/op/key.rs

```diff
@@ -62,6 +64,51 @@ impl GenerateKey {
             KeyType::Ed25519 => self.save_params.save_key(&ed25519_key, "ed22519"),
         }
     }
+
+    /// A test friendly typed key generation for x25519 keys.
+    pub fn generate_x25519(
+        encoding_type: EncodingType,
+        key_file: &Path,
+    ) -> Result<(x25519::PrivateKey, x25519::PublicKey), Error> {
+        let args = format!(
+            "generate --key-type {key_type:?} --key-file {key_file} --encoding {encoding_type:?} --assume-yes",
+            key_type = KeyType::X25519,
+            key_file = key_file.to_str().unwrap(),
+            encoding_type = encoding_type,
+        );
+        let command = GenerateKey::from_iter(args.split_whitespace());
+        command.execute()?;
+        Ok((
+            load_key(key_file, encoding_type)?,
+            load_key(&append_file_extension(key_file, ".pub")?, encoding_type)?,
```

Comment:
nit: I'd maybe move `.pub` to a global and reference it multiple times (so it's obvious what file conventions we're using)